### PR TITLE
Feature: Support for BigPictureStyle notifications from 'imageurl' 

### PR DESCRIPTION
**Pull request by harish2704**
Summary of changes
- If the notification bundle don't have a field called  'imageUrl', Then, nothing special. It will work as normal.
- If the data bundle has a imgeUrl, then,
  - It will fetch the image using Fresco library, Which is already used by React-native
  - Create a notification in BigPictureStyle using fetched Image.

### CODE REVIEW

**Comment by npomfret** [comment]
Could you please review this PR and resolve the conflicts.  I look at merging it after.

**Comment by harish2704** [comment]
Bro, I need some time for this because I am away from react-native based projects.

**Comment by danleveille** [comment]
Can't wait to see this implemented! I'd help out, but it's a little out of my area of expertise.

**Comment by harish2704** [comment]
Hi, I am working on it. Will update you soon

**Comment by harish2704** [comment]
Hi bro, I have merged it. But it is not tested after the merge.
 
Its because, As you may be knowing, React-native version dependency increased from 0.25 to 0.33 since I created the pull request. So I was not able to test it with the original project which I did development before

It will be great idea If some one can add an example "Helloworld" project for this library.

It will help in this kind of situation and also helps any one who tries this library at fist time


In short, for testing, I need to setup an hello-world app with this project. and that will take some more time

**Comment by samin** [comment]
@npomfret no conflicts, any idea when a merge is possible?

**Comment by npomfret** [comment]
Could someone send me a quick note on what's the expected change in behaviour and how I can test this - thanks

**Comment by harish2704** [comment]
Hope you read the  summary section in the pull request. (  see top of the thread )

For testing this new change, you just need to add one field 'imageUrl' in payload bundle. The value should be a publically available image url.

**Comment by npomfret** [comment]
Hi.  Sorry about this, but there are some conflicts.  Are you able to resolve them for me?

File: android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java [hunk 1]
```diff
@@ -128,14 +143,53 @@ public void sendNotificationScheduledCore(Bundle bundle) {
...
+        DataSource<CloseableReference<CloseableImage>> dataSource =
+                imagePipeline.fetchDecodedImage(imageRequest, mContext);
+
+
+        dataSource.subscribe(new BaseBitmapDataSubscriber() {
+            @Override
+            public void onNewResultImpl(@Nullable Bitmap bitmap) {
+                if (bitmap == null) {
+                    Log.d(TAG, "Bitmap data source returned success, but bitmap null.");
+                    sendNotificationWithImage(bundle, null);
```
Comment (npomfret) [inline android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java]: This line and the return statement are not needed.  You can just log the messages and continue.  I'd also consider looking a verbose level.

**Comment by npomfret** [comment]
Hi - apologies for the delay, and thank you for submitting this change.  Struggling to find time to look at PR's on this project. 

Can you please make sure you describe the motivation for this change, and crucially **what is the expected change in behaviour**?  How could someone test it?

I would suggest you ask the authors of other outstanding PR's to test your change in a private fork.  If you can get the thumbs up from a couple of other developers then I'd be happy to merge it.  In exchange you could offer to test theirs maybe..?  That would really help get this project moving.

**Comment by Gurmindermultani** [comment]
I needed this bit in my project. Its still not merged. Any idea when this will get merged?
Thanks in advance.

**Comment by darkmantle** [comment]
Any news on this? Was this finished, or an alternate finished?
